dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -183,6 +190,25 @@ private static void createAliasAndsetUpRelatedIdentifiers(DepositMetadata deposi
     }
 
 
+    /**
+     * extract a digital object identifier (DOI) from a DOI target URL
+     * @param doiUrl digital object identifier
+     * @return the DOI as a string
+     */
+    private static String extractDoiFromDoiUrl(String doiUrl) {
+        // Remove the 'https://doi.org/' etc. prefix from the DOI
+        // e.g. https://doi.org/10.5072/zenodo.372767
+        String doi = doiUrl;
+        try {
+            URI uri = new URI(doiUrl);
+            String[] segments = uri.getPath().split("/");
+            doi = segments[segments.length - 2] + "/" + segments[segments.length - 1];
```

Comment:
```
            URL url = new URL(doiUrl);
            doi = StringUtils.stripStart(uri.getPath(), "/");
```
seems alot simpler in my opinion

Reply:
I have this "https://doi.org/10.5072/zenodo.372767" and need to get this "10.5072/zenodo.372767"